topics/data-science/tutorials/snakemake/tutorial.md

```diff
@@ -0,0 +1,1714 @@
+
+This tutorial is aimed at students and Galaxy community members who might want to convert Snakemake workflows into Galaxy workflows, but need to understand how Snakemake workflows work.
+
+> ### Agenda
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+We're going to go through this tutorial as "the evolution of a pipeline" from a simple bash script that we might run as individual commands on the command line, all the way up to a Snakemake workflow. This should give you some perspective of why the systems exist, and what benefits each system brings.
+
+# Bash
+
+We've set up a simple bash pipeline. It downloads some read files from a website, decompresses them, builds an index for a `genome.fa` we already have, and then does the alignment against that genome.
+
+```bash
+# Downloading our datasets
+wget https://zenodo.org/record/5562251/files/GCA_000017985.1_ASM1798v1_genomic.fna.gz
+wget https://zenodo.org/record/5562251/files/SRR2584866_1.fq.gz
+wget https://zenodo.org/record/5562251/files/SRR2584866_2.fq.gz
+wget https://zenodo.org/record/5562251/files/SRR2589044_1.fq.gz
+wget https://zenodo.org/record/5562251/files/SRR2589044_2.fq.gz
+
+# Generate FastQC Report
+fastqc *.fq
+
+# Run Trimmomatic to trim the bad reads out.
+trimmomatic PE SRR2589044_1.fq.gz SRR2589044_2.fq.gz \
+               SRR2589044_1.trim.fq SRR2589044_1un.trim.fq \
+               SRR2589044_2.trim.fq SRR2589044_2un.trim.fq \
+               SLIDINGWINDOW:4:20 MINLEN:25 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15
+
+trimmomatic PE SRR2584866_1.fq.gz SRR2589044_2.fq.gz \
+               SRR2584866_1.trim.fq SRR2589044_1un.trim.fq \
+               SRR2584866_2.trim.fq SRR2589044_2un.trim.fq \
+               SLIDINGWINDOW:4:20 MINLEN:25 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15
+
+# Generate Updated FastQC Report
+fastqc *.trim.fq
+
+# Generate the genome index
+bwa index GCA_000017985.1_ASM1798v1_genomic.fna
+
+# Align reads to the genome
+bwa mem GCA_000017985.1_ASM1798v1_genomic.fna SRR2589044_1.trim.fq SRR2589044_2.trim.fq | samtools sort -O bam -o SRR2589044.bam
+bwa mem GCA_000017985.1_ASM1798v1_genomic.fna SRR2584866_1.trim.fq SRR2584866_2.trim.fq | samtools sort -O bam -o SRR2584866.bam
+```
+
+This is a fine start to our analysis pipeline, and might simply be summarising
+the commands we executed interactively at the command line.
+
+# Make
+
+Make is a build automation tool that has been around since 1976. It lets you describe build processes, writing down the individual processes which will occur, and then providing you a single entrypoint to run your workflow, a lot like how {SciWMS} work! But importantly it is declarative, rather than imperative which is a big change if you're familiar with programming languages like bash or python.
```

Comment:
is the {SciWMS} a broken link or ?

Reply:
https://training.galaxyproject.org/training-material/news/2021/05/25/abbreviations-tag.html